samtranslator/metrics/method_decorator.py

```diff
@@ -0,0 +1,110 @@
+"""
+Method decorator for execution latency collection
+"""
+import functools
+from datetime import datetime
+from samtranslator.metrics.metrics import DummyMetricsPublisher, Metrics
+from samtranslator.model import Resource
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class MetricsMethodWrapperSingleton:
+    """
+    Keeps the instance of Metrics object.
+    This singleton will be alive until lambda receives shutdown event
+    """
+
+    _METRICS_INSTANCE = None
```

Comment:
I think capital letter are reserved for constants.
This should maybe be lowercase.

Reply:
Yes, but static instances are also using uppercase names (`LOG` for instance). I didn't want this to be accepted as instance variable that is why I went with uppercase.